java/simulator/src/main/java/com/hazelcast/simulator/coordinator/Coordinator.java

```diff
@@ -171,15 +176,31 @@ private void stopTests() {
         LOGGER.info("The following tests failed to complete: " + tests);
     }
 
-    private void startClient() {
-        // todo: should be async to speed things up
+    private void startClient() throws ExecutionException, InterruptedException {
+       ExecutorService executor = Executors.newFixedThreadPool(PARALLEL_CONNECT);
+        List<Future> futures = new ArrayList<>();
         for (AgentData agent : registry.getAgents()) {
+            Future f = executor.submit(() -> {
+                try {
+                    client.connectToAgentBroker(agent.getAddress(), agent.getPublicAddress());
+                } catch (Exception e) {
+                    LOGGER.debug(e.getMessage(), e);
+                    throw new CommandLineExitException("Failed to connect to agent [" + agent.getPublicAddress() + "], "
+                            + "cause [" + e.getMessage() + "]");
+                }
+            });
+            futures.add(f);
+        }
+
+        for (Future future : futures) {
```

Comment:
```suggestion
        List<Future<?>> futures = registry.getAgents().stream().map(agent -> {
            return executor.submit(() -> {
                try {
                    return client.connectToAgentBroker(agent.getAddress(), agent.getPublicAddress());
                } catch (Exception e) {
                    LOGGER.debug(e.getMessage(), e);
                    throw new CommandLineExitException("Failed to connect to agent [" + agent.getPublicAddress() + "], "
                                    + "cause [" + e.getMessage() + "]");
                }
            });
        }).collect(Collectors.toList());

        for (Future<?> future : futures) {
```

Some suggestions:
- Avoiding raw `Future` reference
- Avoid explicit `ArrayList` implementation by using `Stream` collector instead